compare/compare.c

```diff
@@ -593,14 +596,21 @@ int main(int argc, char *argv[])
     ItemList list = {0};
     int result = R_ERROR;
 
-    int scale = 1;
+    int scale_x = 1, scale_y = 1;
     if (pos[SCALE]) {
         const char *arg = argv[pos[SCALE]];
-        if (arg[0] < '1' || arg[0] > '8' || arg[1]) {
+        if (arg[0] < '1' || arg[0] > '8' || (arg[1] && arg[1] != 'x')) {
             printf("Invalid scale value, should be 1-8!\n");
             goto end;
         }
-        scale = arg[0] - '0';
+        scale_x = scale_y = arg[0] - '0';
+        if (arg[1]) {
+            if (arg[2] < '1' || arg[2] > '8' || arg[3]) {
+                printf("Invalid scale value, should be 1-8!\n");
```

Comment:
Maybe it's better to move scale parsing into separate function to eliminate copy-paste? I.e.
```c
        if (!parse_scale(argv[pos[SCALE]], &scale_x, &scale_y)) {
            printf("Invalid scale value, should be 1-8!\n");
            goto end;
        }
```

Reply:
Done.

By the way, does the control flow within the block (now the function) seem fine to you, or would you prefer it to be tweaked in some way? I’m finding myself not quite liking any of the ways I can write it.

Reply:
Hmm... maybe something like
```c
static bool parse_scale(const char *arg,  int *scale_x, int *scale_y)
{
    if (arg[0] < '1' || arg[0] > '8')
        return false;
    *scale_x = *scale_y = arg[0] - '0';
    if (!arg[1])
        return true;
    if (arg[1] != 'x' || arg[2] < '1' || arg[2] > '8' || arg[3])
        return false;
    *scale_y = arg[2] - '0';
    return true;
}
```

Reply:
Applied, thanks.